pkg/util/protoutil/marshaler.go

```diff
@@ -101,3 +101,8 @@ func (e *protoEncoder) Encode(v interface{}) error {
 	}
 	return errors.Errorf("unexpected type %T does not implement %s", v, typeProtoMessage)
 }
+
+// Delimiter implements gwruntime.Delimited.
+func (*ProtoPb) Delimiter() []byte {
+	return []byte("\n")
```

Comment:
Encoded protobufs may contain `\n` characters. I don't think there is any character (or sequence) that can safely be used as a delimiter here - protobufs are generally used in protocols with length-prefixed frames instead of delimiters.

Reply:
That's a great point. Not sure there's too much we can do here though since it'll default to `\n` if we don't use it ourselves. I'll remove this again so at least it's the grpc-gateway that's wrong.

Reply:
I've removed it.